flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/DefaultCompletedCheckpointStore.java

```diff
@@ -95,18 +86,14 @@ public DefaultCompletedCheckpointStore(
             int maxNumberOfCheckpointsToRetain,
             StateHandleStore<CompletedCheckpoint, R> stateHandleStore,
             CheckpointStoreUtil completedCheckpointStoreUtil,
+            Collection<CompletedCheckpoint> completedCheckpoints,
             Executor executor) {
-
         checkArgument(maxNumberOfCheckpointsToRetain >= 1, "Must retain at least one checkpoint.");
-
         this.maxNumberOfCheckpointsToRetain = maxNumberOfCheckpointsToRetain;
-
         this.checkpointStateHandleStore = checkNotNull(stateHandleStore);
-
         this.completedCheckpoints = new ArrayDeque<>(maxNumberOfCheckpointsToRetain + 1);
```

Comment:
not necessary a comment related to this PR's changes but `maxNumberOfCheckpointsToRetain` does not enforce `completedCheckpoints` from growing. is that something that needs to be addressed?

Reply:
Also we can not enforce this when constructing a new CompletedCheckpointStore, because we've not yet registered completed checkpoints with a SharedStateRegistry, so this might break incremental checkpoints.

There is actually no change from the previous behavior.